core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_MDRange.hpp

```diff
@@ -0,0 +1,457 @@
+    const ptrdiff_t begin_0 = offset[0];
+    ptrdiff_t end_0         = begin_0 + policy.m_tile[0];
+    end_0 = end_0 < policy.m_upper[0] ? end_0 : policy.m_upper[0];
+
+    const ptrdiff_t begin_1 = offset[1];
+    ptrdiff_t end_1         = begin_1 + policy.m_tile[1];
+    end_1 = end_1 < policy.m_upper[1] ? end_1 : policy.m_upper[1];
+
+    const ptrdiff_t begin_2 = offset[2];
+    ptrdiff_t end_2         = begin_2 + policy.m_tile[2];
+    end_2 = end_2 < policy.m_upper[2] ? end_2 : policy.m_upper[2];
+
+    const ptrdiff_t begin_3 = offset[3];
+    ptrdiff_t end_3         = begin_3 + policy.m_tile[3];
+    end_3 = end_3 < policy.m_upper[3] ? end_3 : policy.m_upper[3];
+
+    const ptrdiff_t begin_4 = offset[4];
+    ptrdiff_t end_4         = begin_4 + policy.m_tile[4];
+    end_4 = end_4 < policy.m_upper[4] ? end_4 : policy.m_upper[4];
+
+    const ptrdiff_t begin_5 = offset[5];
+    ptrdiff_t end_5         = begin_5 + policy.m_tile[5];
+    end_5 = end_5 < policy.m_upper[5] ? end_5 : policy.m_upper[5];
+
+#pragma omp for collapse(6)
+    for (ptrdiff_t i0 = begin_0; i0 < end_0; i0++)
+      for (ptrdiff_t i1 = begin_1; i1 < end_1; i1++)
+        for (ptrdiff_t i2 = begin_2; i2 < end_2; i2++)
+          for (ptrdiff_t i3 = begin_3; i3 < end_3; i3++)
+            for (ptrdiff_t i4 = begin_4; i4 < end_4; i4++)
+              for (ptrdiff_t i5 = begin_5; i5 < end_5; i5++)
+                functor(i0, i1, i2, i3, i4, i5);
+  }
+
+  inline ParallelFor(const FunctorType& arg_functor, Policy arg_policy)
+      : m_functor(arg_functor), m_policy(arg_policy) {}
+};
+
+}  // namespace Impl
+}  // namespace Kokkos
+
+//----------------------------------------------------------------------------
+//----------------------------------------------------------------------------
+
+namespace Kokkos {
+namespace Impl {
+
+template <class FunctorType, class ReducerType, class PointerType,
+          class ValueType, class... PolicyArgs>
+struct ParallelReduceSpecialize<FunctorType,
+                                Kokkos::MDRangePolicy<PolicyArgs...>,
+                                ReducerType, PointerType, ValueType, 0, 0> {
+  typedef Kokkos::RangePolicy<PolicyArgs...> PolicyType;
+  template <class TagType>
+  inline static
+      typename std::enable_if<std::is_same<TagType, void>::value>::type
+      execute_impl(const FunctorType& f, const PolicyType& p,
+                   PointerType result_ptr) {
+    OpenMPTargetExec::verify_is_process(
+        "Kokkos::Experimental::OpenMPTarget parallel_for");
+    OpenMPTargetExec::verify_initialized(
+        "Kokkos::Experimental::OpenMPTarget parallel_for");
+    const typename PolicyType::member_type begin = p.begin();
+    const typename PolicyType::member_type end   = p.end();
+
+    ValueType result = ValueType();
+#pragma omp target teams distribute parallel for num_teams(512) map(to:f) map(tofrom:result) reduction(+: result)
+    for (int i = begin; i < end; i++) f(i, result);
```

Comment:
Do we check somewhere that `f` is a sum reduction?

Reply:
Yeah the 0,0 in the specialization says no custom reducer was defined ...